pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -338,55 +338,104 @@ func (config *V2RegistriesConf) postProcess() error {
 }
 
 // ConfigPath returns the path to the system-wide registry configuration file.
+// Deprecated: This API only has support one registries.conf from system-wide or per-user.
+// To get the support for file and directory override, we added the new
+// ConfigurationSourceDescription API. The new API should be used and this API is kept to
+// maintain backward compatibility.
 func ConfigPath(ctx *types.SystemContext) string {
-	if ctx != nil && ctx.SystemRegistriesConfPath != "" {
-		return ctx.SystemRegistriesConfPath
-	}
-
-	userRegistriesFilePath := filepath.Join(homedir.Get(), userRegistriesFile)
-	if _, err := os.Stat(userRegistriesFilePath); err == nil {
-		return userRegistriesFilePath
-	}
-
-	if ctx != nil && ctx.RootForImplicitAbsolutePaths != "" {
-		return filepath.Join(ctx.RootForImplicitAbsolutePaths, systemRegistriesConfPath)
-	}
-
-	return systemRegistriesConfPath
+	wrapper := newConfigWrapper(ctx)
+	return wrapper.configPath
 }
 
 // ConfigDirPath returns the path to the system-wide directory for drop-in
 // registry configuration files.
+// Deprecated: This API only has support one directory from system-wide or per-user.
+// To get the support for file and directory override, we added the new
+// ConfigurationSourceDescription API. The new API should be used and this API is kept to
+// maintain backward compatibility.
 func ConfigDirPath(ctx *types.SystemContext) string {
-	if ctx != nil && ctx.SystemRegistriesConfDirPath != "" {
-		return ctx.SystemRegistriesConfDirPath
-	}
-
-	userRegistriesDirPath := filepath.Join(homedir.Get(), userRegistriesDir)
-	if _, err := os.Stat(userRegistriesDirPath); err == nil {
-		return userRegistriesDirPath
-	}
-
-	if ctx != nil && ctx.RootForImplicitAbsolutePaths != "" {
-		return filepath.Join(ctx.RootForImplicitAbsolutePaths, systemRegistriesConfDirPath)
+	wrapper := newConfigWrapper(ctx)
+	if wrapper.userConfigDirPath != "" {
```

Comment:
To the little extent it matters, the documentation says “to the system-wide directory”, so this should probably prefer the system path if both are set.